### ISSUE TITLE
Support for parameters in a file/other format?

### ISSUE THREAD

**Issue by isurulucky** [deislabs/cnab-spec#67]
In the current CNAB spec, the parameters are defined inline with the bundle.json file. These parameters can either be exposed to the Invocation Image either as a environment variable or can be written to a file. However, for an application component which is depending on a defined configuration model in the form of a file (json/xml, etc.), the configuration values being exposed as environment variables, or being written to individual files might not be helpful. That would mean the particular component would have to be modified to fit the CNAB model. 

In addition to the how parameters are currently defined, if a configuration element can be defined in the parameters file, with the relevant information on how to validate it (type, min, max, etc., similar to how the current parameters are implemented), and be looked up from a file source, it could be useful for some cases. After validation of the parameters in the source file, the same can be written to a target path (ex.: file path in the container when using docker driver), which can be read from the particular application component. 

At a high level, this would require changes to the parameters section as shown below:

```json
"parameters": {
    "backend_port" : {
        "type" : "int",
        "defaultValue": 80,
        "minValue": 10,
        "maxValue": 10240,
        "metadata": {
            "description": "The port that the backend will listen on"
        },
        "source": {  
             "file": "~/dev/config/params.conf",
             "lookup": "$.server.http.port[0]"
        }
    },
    "destination": {
        "file": "/home/config/params.conf"
    }
}
```
Note the new `source` block which tells where to find this parameter from the local file system, as well as the `lookup` parameter which can be used to extract this particular parameter for validation purposes. In addition, a new `destination` block is added to the top level of parameters, which include parameter `file`, denoting the path where this config will be written to in the target application component, which can be then read at the runtime. 

This model can be extended to support sources other than files as well. 

Please do share your feedback on this approach, whether this would be useful/feasible.

**Comment by technosophos** [comment]
What is "local filesystem" in this context?

- The system from which a client is run (Assuming there is a client)
- The host running the invocation image
- The inside filesystem of the invocation image
- Other

**Comment by technosophos** [comment]
I talked to @vdice about this issue a little bit today. I think I understand better.

If the desire is to take a file from the client's local system and send it into a CNAB bundle, this is accomplished already in `duffle` using the a flag:

````
  -i, --set-file stringArray      Set individual parameters from file content as NAME=SOURCE-PATH pairs
```

vdice also suggested that perhaps we should have an option for somehow attaching volumes to the invocation image. I need to think through that scenario a little more.

**Comment by isurulucky** [comment]
Thanks @technosophos, yes the idea is similar to what you have described. Attaching volumes would be a good option opposed to handling the complexity of selectively dumping configurations from files in the local file system to the invocation image.

**Comment by technosophos** [comment]
Issue #114 is now converging on a solution that I think will handle this.

**Comment by technosophos** [comment]
This was resolved with #114
